source/common/upstream/cluster_manager_impl.cc

```diff
@@ -419,9 +419,12 @@ void ClusterManagerImpl::createOrUpdateThreadLocalCluster(ClusterData& cluster)
               ENVOY_LOG(debug, "adding TLS cluster {}", new_cluster->name());
             }
 
-            cluster_manager.thread_local_clusters_[new_cluster->name()].reset(
-                new ThreadLocalClusterManagerImpl::ClusterEntry(cluster_manager, new_cluster,
-                                                                thread_aware_lb_factory));
+            auto thread_local_cluster = new ThreadLocalClusterManagerImpl::ClusterEntry(
+                cluster_manager, new_cluster, thread_aware_lb_factory);
+            cluster_manager.thread_local_clusters_[new_cluster->name()].reset(thread_local_cluster);
```

Comment:
I would make this a `make_unique` and then access via `*` below still.

Reply:
I did, then reverted based on Matt's comment about double deletes.